pfhedge/nn/modules/bs/european.py

```diff
@@ -148,6 +148,61 @@ def gamma(
 
         return gamma
 
+    def vega(
+        self, log_moneyness: Tensor, time_to_maturity: Tensor, volatility: Tensor
+    ) -> Tensor:
+        """Returns vega of the derivative.
+
+        Args:
+            log_moneyness: (torch.Tensor): Log moneyness of the underlying asset.
+            time_to_maturity (torch.Tensor): Time to expiry of the option.
+            volatility (torch.Tensor): Volatility of the underlying asset.
+
+        Shape:
+            - log_moneyness: :math:`(N, *)` where
+              :math:`*` means any number of additional dimensions.
+            - time_to_maturity: :math:`(N, *)`
+            - volatility: :math:`(N, *)`
+            - output: :math:`(N, *)`
+
+        Returns:
+            torch.Tensor
+        """
+        s, t, v = broadcast_all(log_moneyness, time_to_maturity, volatility)
+        price = self.strike * s.exp()
+        vega = npdf(d1(s, t, v)) * price * t.sqrt()
+
+        return vega
+
+    def theta(
+        self, log_moneyness: Tensor, time_to_maturity: Tensor, volatility: Tensor
+    ) -> Tensor:
+        """Returns theta (yearly) of the derivative.
```

Comment:
just a remark (not requiring any correction): unit of theta is not a priori; determined by that of volatility and time_to_maturity